File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        i = 0
        ans = []
        while i < len(nums):
            index = abs(nums[i])
            #print(index)
            if nums[index-1] < 0:
                ans.append(index)   
            else:
                nums[index-1] = -nums[index-1]
            i+=1
        return ans
            
                
        

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> ans = new ArrayList<Integer>();
        for(int i = 0; i < nums.length; i++){
            int curr = Math.abs(nums[i])-1;
            //System.out.println(curr);
            if (nums[curr] < 0){
                ans.add(curr+1);
            }
            else{
                nums[curr] *= -1;
            }
        }
        for(int i = 0; i < nums.length; i++){
            nums[i] = Math.abs(nums[i]);
        }
        return ans;
        
    }
}